`from_chars` is not a member of `std`

**Describe the bug**
```
2026-04-20T06:42:13.2427986Z -- Build files have been written to: /__w/ros2_control/ros2_control/ros_ws/build/hardware_interface
2026-04-20T06:42:13.2428933Z [  3%] Generating hardware_interface/version.h
2026-04-20T06:42:13.2429710Z [  3%] Built target ament_generate_version_header__hardware_interface
2026-04-20T06:42:13.2430669Z [  6%] Building CXX object CMakeFiles/test_macros.dir/test/test_macros.cpp.o
2026-04-20T06:42:13.2431422Z [  9%] Linking CXX executable test_macros
2026-04-20T06:42:13.2431950Z [  9%] Built target test_macros
2026-04-20T06:42:13.2432793Z [ 12%] Building CXX object CMakeFiles/hardware_interface.dir/src/component_parser.cpp.o
2026-04-20T06:42:13.2434088Z [ 15%] Building CXX object CMakeFiles/hardware_interface.dir/src/hardware_component_interface.cpp.o
2026-04-20T06:42:13.2435358Z [ 18%] Building CXX object CMakeFiles/hardware_interface.dir/src/hardware_component.cpp.o
2026-04-20T06:42:13.2436555Z [ 21%] Building CXX object CMakeFiles/hardware_interface.dir/src/resource_manager.cpp.o
2026-04-20T06:42:13.2437682Z [ 24%] Building CXX object CMakeFiles/hardware_interface.dir/src/lexical_casts.cpp.o
2026-04-20T06:42:13.2440745Z /__w/ros2_control/ros2_control/ros_ws/src/v4zl7emx1p/ros2_control/hardware_interface/src/lexical_casts.cpp: In function ‘std::optional<double> hardware_interface::impl::stod(const std::string&)’:
2026-04-20T06:42:13.2443660Z /__w/ros2_control/ros2_control/ros_ws/src/v4zl7emx1p/ros2_control/hardware_interface/src/lexical_casts.cpp:48:34: error: ‘from_chars’ is not a member of ‘std’
2026-04-20T06:42:13.2445343Z    48 |   const auto parse_result = std::from_chars(s.data(), s.data() + s.size(), result_value);
2026-04-20T06:42:13.2446185Z       |                                  ^~~~~~~~~~
2026-04-20T06:42:13.2448725Z /__w/ros2_control/ros2_control/ros_ws/src/v4zl7emx1p/ros2_control/hardware_interface/src/lexical_casts.cpp: In function ‘std::optional<float> hardware_interface::impl::stof(const std::string&)’:
2026-04-20T06:42:13.2451505Z /__w/ros2_control/ros2_control/ros_ws/src/v4zl7emx1p/ros2_control/hardware_interface/src/lexical_casts.cpp:74:34: error: ‘from_chars’ is not a member of ‘std’
2026-04-20T06:42:13.2453300Z    74 |   const auto parse_result = std::from_chars(s.data(), s.data() + s.size(), result_value);
2026-04-20T06:42:13.2454265Z       |                                  ^~~~~~~~~~
2026-04-20T06:42:13.2455595Z gmake[2]: *** [CMakeFiles/hardware_interface.dir/build.make:132: CMakeFiles/hardware_interface.dir/src/lexical_casts.cpp.o] Error 1
2026-04-20T06:42:13.2456795Z gmake[2]: *** Waiting for unfinished jobs....
2026-04-20T06:42:13.2458045Z gmake[1]: *** [CMakeFiles/Makefile2:161: CMakeFiles/hardware_interface.dir/all] Error 2
2026-04-20T06:42:13.2458944Z gmake: *** [Makefile:146: all] Error 2
2026-04-20T06:42:13.2459366Z ---
2026-04-20T06:42:13.2459743Z --- stderr: hardware_interface
2026-04-20T06:42:13.2461994Z /__w/ros2_control/ros2_control/ros_ws/src/v4zl7emx1p/ros2_control/hardware_interface/src/lexical_casts.cpp: In function ‘std::optional<double> hardware_interface::impl::stod(const std::string&)’:
2026-04-20T06:42:13.2464866Z /__w/ros2_control/ros2_control/ros_ws/src/v4zl7emx1p/ros2_control/hardware_interface/src/lexical_casts.cpp:48:34: error: ‘from_chars’ is not a member of ‘std’
2026-04-20T06:42:13.2466597Z    48 |   const auto parse_result = std::from_chars(s.data(), s.data() + s.size(), result_value);
2026-04-20T06:42:13.2467449Z       |                                  ^~~~~~~~~~
2026-04-20T06:42:13.2470099Z /__w/ros2_control/ros2_control/ros_ws/src/v4zl7emx1p/ros2_control/hardware_interface/src/lexical_casts.cpp: In function ‘std::optional<float> hardware_interface::impl::stof(const std::string&)’:
2026-04-20T06:42:13.2472925Z /__w/ros2_control/ros2_control/ros_ws/src/v4zl7emx1p/ros2_control/hardware_interface/src/lexical_casts.cpp:74:34: error: ‘from_chars’ is not a member of ‘std’
2026-04-20T06:42:13.2474596Z    74 |   const auto parse_result = std::from_chars(s.data(), s.data() + s.size(), result_value);
2026-04-20T06:42:13.2475619Z       |                                  ^~~~~~~~~~
2026-04-20T06:42:13.2476923Z gmake[2]: *** [CMakeFiles/hardware_interface.dir/build.make:132: CMakeFiles/hardware_interface.dir/src/lexical_casts.cpp.o] Error 1
2026-04-20T06:42:13.2478439Z gmake[2]: *** Waiting for unfinished jobs....
2026-04-20T06:42:13.2479395Z gmake[1]: *** [CMakeFiles/Makefile2:161: CMakeFiles/hardware_interface.dir/all] Error 2
2026-04-20T06:42:13.2480248Z gmake: *** [Makefile:146: all] Error 2
2026-04-20T06:42:13.2480689Z ---
2026-04-20T06:42:13.2481211Z Failed   <<< hardware_interface [31.2s, exited with code 2]
```

**To Reproduce**

- https://github.com/ros-controls/ros2_control/issues/3232
- https://github.com/ros-controls/topic_based_hardware_interfaces/issues/116
- https://github.com/ros-controls/ros2_controllers/issues/2319

--- a/hardware_interface/src/lexical_casts.cpp
+++ b/hardware_interface/src/lexical_casts.cpp
@@ -14,6 +14,7 @@
 
 #include <algorithm>
 #include <cctype>
+#include <charconv>
 #include <cmath>
 #include <cstdint>
 #include <locale>

